setup-buildengine.sh

```diff
@@ -403,10 +399,13 @@ while [[ ${1:-} ]]; do
             OPT_TOOLING=$(basename $1); shift
             ;;
         -ta | --target-arm ) shift
-            OPT_TARGET_ARM=$(basename $1); shift
+            OPT_TARGET_FILENAMES[armv7hl]="$(basename "$1")"; shift
+            ;;
+        -tA | --target-aarch64 ) shift
+            OPT_TARGETS[aarch64]="$(basename "$1")"; shift
             ;;
         -ti | --target-i486 ) shift
-            OPT_TARGET_I486=$(basename $1); shift
+            OPT_TARGET_FILENAMES[i486]="$(basename "$1")"; shift
             ;;
```

Comment:
```suggestion
        --target-* )
            OPT_TARGETS[${1#--target-}]=$2
            shift 2
            ;;
```

The basename use looks buggy.

Reply:
It is to avoid word splitting in case $1 has space.

Reply:
Hm, what do you mean more exactly? I mean that discarding the path with `basename` renders the argument useless unless it already was a base name, in which case it is a no-op. Maybe the original author wanted to avoid possible `./` prefix when printing the summary, but in that case they should use `realpath -f` instead. So IIUC, the argument would be best used as is and the same should be applied to tooling path.

Reply:
Note that you do not need quotes in variable assignment unless the expression itself contains characters that need quoting. So `x=$y` and `x=$(y "$z")`, as well as `x="foo bar"/$y` works correctly no matter what the right side evaluates to.

Reply:
In the latest version I still see the -arch part of the options hardcoded.